torch/optim/sgd.py

```diff
@@ -204,8 +204,14 @@ def sgd(params: List[Tensor],
     """
 
     if foreach is None:
-        # Placeholder for more complex foreach logic to be added when value is not set
-        foreach = False
+        # why must we handle momentum_buffer_list separately?
```

Comment:
Is it when tracing or scripting? If the later, than can we just guard this all statement with a is_scripting ?

Reply:
The test passes now (so it is scripting in the test) but should we worry about tracing requirements?